### PR TITLE
Cable modeling Implementation

### PR DESCRIPTION
**Pull request by gmalinve**
In this PR:
- Features to create, update, remove or add cables have been implemented.
- New tests
- json files to test
- aedt file to test

### CODE REVIEW

File: pyaedt/modules/CableModeling.py [hunk 1]
```diff
@@ -0,0 +1,609 @@
+import itertools
+import json
+import os
+
+from pyaedt.generic.general_methods import generate_unique_name
+from pyaedt.generic.LoadAEDTFile import load_entire_aedt_file
+
+
+class Cable:
+    def __init__(self, app, json_file_name=None, working_dir=None):
```
Comment (MaxJPRey) [inline pyaedt/modules/CableModeling.py:10]: @gmalinve Please add some docstrings for the class here. Something like:

```
"""Description
Parameters
------------
my_parameter: list
    Parameter description.

Examples
----------
    >>>some example
"""
```

File: pyaedt/modules/CableModeling.py [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+                twisted_pair_cables_list.append(self.cable_definitions.get("TwistedPairCable"))
+            else:
+                twisted_pair_cables_list = self.cable_definitions.get("TwistedPairCable")
+            for x in twisted_pair_cables_list:
+                self.existing_twisted_pair_cables_names.append(x["TwistedPairAttribs"]["Name"])
+
+        if json_file_name:
+            self._init_from_json(json_file_name)
+
+    def create_cable(self):
```
Comment (MaxJPRey) [inline pyaedt/modules/CableModeling.py:55]: Same here, please add docstrings for all methods.

**Comment by Samuelopez-ansys** [comment]
@maxcapodi78 What do you think? Should it be in Analysis.py or Modeler.py?

File: pyaedt/modules/CableModeling.py [hunk 3]
```diff
@@ -0,0 +1,663 @@
...
+
+class Cable:
+    """Contains all common Cable features.
+
+    Parameters
+    ----------
+    app : :class:`pyaedt.hfss.Hfss`
+    json_file_name : str , optional
+        Path of the json file where the cable information are saved.
+    working_dir : str , optional
```
Comment (MaxJPRey) [inline pyaedt/modules/CableModeling.py:17]:
```suggestion
    working_dir : str, optional
```

File: pyaedt/modules/CableModeling.py [hunk 4]
```diff
@@ -0,0 +1,663 @@
...
+from pyaedt.generic.LoadAEDTFile import load_entire_aedt_file
+
+
+class Cable:
+    """Contains all common Cable features.
+
+    Parameters
+    ----------
+    app : :class:`pyaedt.hfss.Hfss`
+    json_file_name : str , optional
```
Comment (MaxJPRey) [inline pyaedt/modules/CableModeling.py:15]:
```suggestion
    json_file_name : str, optional
```

File: pyaedt/modules/CableModeling.py [hunk 5]
```diff
@@ -0,0 +1,663 @@
...
+                        )
+                        raise ValueError(msg)
+                    self.straight_wire_insulation_thickness = cable_st_wire_properties["StWireParams"]["InsThickness"]
+                    if self._app.materials.material_keys.get(
+                        cable_st_wire_properties["StWireParams"]["InsMaterial"].lower()
+                    ):
+                        self.insulation_material = cable_st_wire_properties["StWireParams"]["InsMaterial"]
+                    else:
+                        msg = "material provided doesn't exist."
+                        raise ValueError(msg)
```
Comment (MaxJPRey) [inline pyaedt/modules/CableModeling.py:606]:
```suggestion
                        raise ValueError("Material provided doesn't exist.")
```

File: pyaedt/modules/CableModeling.py [hunk 6]
```diff
@@ -0,0 +1,663 @@
...
+                        index = self.existing_straight_wire_cables_ids.index(
+                            cable_twisted_pair_properties["TwistedPairParams"]["StraightWireCableID"]
+                        )
+                        self.assign_cable_to_twisted_pair = self.existing_straight_wire_cables_names[index]
+                    elif not cable_twisted_pair_properties["TwistedPairParams"]["StraightWireCableID"]:
+                        msg = "no cable id provided. Please enter at least one valid cable id."
+                        raise ValueError(msg)
+                    else:
+                        msg = (
+                            "cable provided doesn't exist in the current project. Please provide an existing cable id."
```
Comment (MaxJPRey) [inline pyaedt/modules/CableModeling.py:633]:
```suggestion
                            "Cable provided doesn't exist in the current project. Please provide an existing cable id."
```

File: pyaedt/modules/CableModeling.py [hunk 7]
```diff
@@ -0,0 +1,663 @@
...
+                            insulation_type_options = ["Thin Wall", "Ultra-Thin Wall"]
+                        elif self.wire_type in ["0.5", "0.75", "1", "1.5", "2", "2.5"]:
+                            insulation_type_options = ["Thick Wall", "Thin Wall", "Ultra-Thin Wall"]
+                        if self.wire_type in ["3", "4", "5", "6", "10", "16", "25"]:
+                            insulation_type_options = ["Thick Wall", "Thin Wall"]
+                        elif self.wire_type in ["35", "50", "70", "95", "120"]:
+                            insulation_type_options = ["Thick Wall"]
+                    else:
+                        msg = (
+                            "wire type not valid. Available options are: 0.13, 0.22, 0.35, 0.5, 0.75, 1, 1.5, 2, 2.5, "
```
Comment (MaxJPRey) [inline pyaedt/modules/CableModeling.py:579]:
```suggestion
                            "Wire type not valid. Available options are: 0.13, 0.22, 0.35, 0.5, 0.75, 1, 1.5, 2, 2.5, "
```

**Review by maxcapodi78 — APPROVED** [review]
Good implementation start.

**Review by MaxJPRey — APPROVED** [review]
That is great. Thanks @gmalinve .

**Pull request merged**